mongodbatlas/resource_mongodbatlas_online_archive.go

```diff
@@ -564,38 +562,36 @@ func mapSchedule(d *schema.ResourceData) *matlas.OnlineArchiveSchedule {
 	}
 
 	if len(scheduleTFConfigList) == 0 {
-		return &matlas.OnlineArchiveSchedule{
-			Type: scheduleTypeDefault,
-		}
+		return scheduleInput
 	}
 
 	scheduleTFConfig := scheduleTFConfigList[0].(map[string]interface{})
-	scheduleInput = &matlas.OnlineArchiveSchedule{
+	scheduleInput = &admin.OnlineArchiveSchedule{
 		Type: scheduleTFConfig["type"].(string),
 	}
 
 	if endHour, ok := scheduleTFConfig["end_hour"].(int); ok {
-		scheduleInput.EndHour = pointy.Int32(int32(endHour))
+		scheduleInput.EndHour = pointy.Int(endHour)
 	}
 
 	if endMinute, ok := scheduleTFConfig["end_minute"].(int); ok {
-		scheduleInput.EndMinute = pointy.Int32(int32(endMinute))
+		scheduleInput.EndMinute = pointy.Int(endMinute)
 	}
 
 	if startHour, ok := scheduleTFConfig["start_hour"].(int); ok {
-		scheduleInput.StartHour = pointy.Int32(int32(startHour))
+		scheduleInput.StartHour = pointy.Int(startHour)
 	}
 
 	if startMinute, ok := scheduleTFConfig["start_minute"].(int); ok {
-		scheduleInput.StartMinute = pointy.Int32(int32(startMinute))
+		scheduleInput.StartMinute = pointy.Int(startMinute)
 	}
 
-	if dayOfWeek, ok := scheduleTFConfig["day_of_week"].(int); ok {
-		scheduleInput.DayOfWeek = int32(dayOfWeek)
+	if dayOfWeek, ok := scheduleTFConfig["day_of_week"].(int); ok && dayOfWeek != 0 { // needed to verify attribute is actually defined
+		scheduleInput.DayOfWeek = pointy.Int(dayOfWeek)
 	}
 
-	if dayOfMonth, ok := scheduleTFConfig["day_of_month"].(int); ok {
-		scheduleInput.DayOfMonth = int32(dayOfMonth)
+	if dayOfMonth, ok := scheduleTFConfig["day_of_month"].(int); ok && dayOfMonth != 0 {
+		scheduleInput.DayOfMonth = pointy.Int(dayOfMonth)
```

Comment:
only setting value when `!= 0` to preserve same behaviour. If this check is not added, property with default value 0 is always sent and causes errors in API.